lib/App/Rakubrew/Shell.pm

```diff
@@ -163,7 +163,7 @@ sub get_completions {
     my $index = shift;
     my @words = @_;
 
-    if ($index == 0) {
+    if ($index <= 0) { # if @words is empty then $index == -1
```

Comment:
By definition the passed index should be the 0-based index of the word to be completed. That can never be 0, as there is no -1th word. :-)
In what situations is a -1 passed? I'd say lets better fix this on the caller side.

Reply:
I considered fixing it on the caller side, but the point is the value comes from shell hook `completions` where, if no arguments passed, index becomes -1 simply for the reason there is no element 0. And, to my view, it's more logical to have -1 indicate "nothing" rather than have it indicate '1 element' and then check if it's actually true.

Also, handling on the caller side would require copy-paste code in basically every shell hook implementation. I don't like the copying from `Zsh` into `Fish` already, but it was too late to find a better way.

Reply:
I think all the other shells should be fine. It's only fish that needs this fix. I'm a bit undecided.

I guess I'm OK with the change.

Can you then change the comment of the function to state that it also accepts -1 which also means to complete the first word?